.github/workflows/testing.yml

```
@@ -0,0 +1,45 @@
+permissions:
+    contents: read
+
+jobs:
+    build:
+        runs-on: ubuntu-latest
+
+        steps:
+        - uses: actions/checkout@v4
+
+        - name: Set up Python 3.12
+          uses: actions/setup-python@v3
```

actionlint (1.7.4)

the runner of "actions/setup-python@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)